src/commands/initialize-broadcast.js

```diff
@@ -1,42 +1,48 @@
 import State from '../state'
-import { logger } from '../helpers/constants'
+// import { logger } from '../helpers/constants'
 
 class InitializeBroadcast {
-  constructor(broadcastService) {
+  constructor({ broadcastService, messageService }) {
     this.broadcastService = broadcastService
+    this.messageService = messageService
     this.commandString = '/broadcast'
   }
 
-  shouldExecute(messageService) {
-    if (messageService.getCommand() === this.commandString) {
+  shouldExecute() {
+    if (this.messageService.getCommand() === this.commandString) {
       return true
     }
     return false
   }
 
-  execute(messageService) {
-    this.broadcastService.setMessage(messageService.getArgument())
-    this.broadcastService.setUserEmail(messageService.getUserEmail())
-    this.broadcastService.setVGroupID(messageService.getVGroupID())
+  execute() {
+    const {
+      argument,
+      // message,
+      userEmail,
+      vGroupID,
+    } = this.messageService.getMessageData()
+    console.log({ vGroupID })
+    this.broadcastService.setMessage(argument)
+    this.broadcastService.setUserEmail(userEmail)
+    this.broadcastService.setVGroupID(vGroupID)
     this.broadcastService.setTTL('')
     this.broadcastService.setBOR('')
     this.broadcastService.setSentByFlag(true)
-    logger.debug(`Here are all the things: ${messageService.getArgument()}`)
+    this.messageService.setUserCurrentState({ currentState: State.ASK_FOR_ACK }) // change to broacastservice?
+
+    console.log('initbroadcast')
+    console.log(this.broadcastService)
     let reply = 'Would you like to ask the recipients for an acknowledgement?'
-    let state = State.ASK_FOR_ACK
     // TODO check for undefined??
-    if (
-      messageService.getArgument() === undefined ||
-      messageService.getArgument() === '' ||
-      messageService.getArgument().length === 0
-    ) {
+    if (!argument) {
```

Comment:
Is this sufficient I feel like it misses a case

Reply:
yes, message service gives data back as null if nothing is there.